python/tvm/relay/op/contrib/ethosu.py

```diff
@@ -650,12 +650,33 @@ def is_valid(self):
         """
         This function checks whether AvgPool2D has compatible attributes with the NPU
         """
+
+        def check_extra_strides():
+            """
+            This function checks whether AvgPool2D could be legalized as ethosu_pooling
+            supported by the NPU.
+            We consider only specific case: when there is no AvgPool2D padding, the spatial
+            dimensions of ifm and the shape of pooling are equal, but stride size exceed 3
+            by any of dimensions, e.g:
+            ifm: (1, 8, 8, _), strides: (8, 8), pool_shape: (8, 8)
+            ifm: (1, 25, 5, _), strides: (25, 5), pool_shape: (25, 5)
+            """
+            if self.pooling_type != "AVG":
+                return False
+            if list(self.padding) != [0, 0, 0, 0]:
+                return False
+            if [self.ifm.shape[1], self.ifm.shape[2]] != list(self.pool_shape):
+                return False
+            if list(self.strides) != list(self.pool_shape):
```

Comment:
Why is this condition necessary? To me it sounds like this patch supports a case where the kernel is the same shape as ifm, so the stride value doesn't matter since we can't move the kernel anyway?

Reply:
Yes, you're right, now I see redundancy as well.